src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,125 @@
+package duke.main;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The class that saves and loads data.
+ */
+public class Storage {
+    private final String filepath;
+
+    /**
+     * Initialisation of the Storage class which requires a filepath to save and load data from.
+     *
+     * @param filepath The relative filepath from where Duke was ran, to save and load data from
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Loads data using the filepath given.
+     *
+     * @param ui Needed to display the outcome to the user
+     * @return Returns the saved TaskList, if there is any
+     */
+    public TaskList loadData(UI ui) {
```

Comment:
Similarly to parse in Parser.java, maybe you can try to refactor loadData by extracting the methods.